pylint/pyreverse/diagrams.py

```diff
@@ -28,59 +43,73 @@ def __init__(self, from_object, to_object, relation_type, name=None):
 class DiagramEntity(Figure):
     """A diagram object, i.e. a label associated to an astroid node."""
 
-    def __init__(self, title="No name", node=None):
+    default_shape = ""
+
+    def __init__(self, title: str = "No name", node: NodeNG = None) -> None:
         super().__init__()
         self.title = title
-        self.node = node
+        self.node: NodeNG = node if node else NodeNG()
+        self.shape = self.default_shape
 
 
 class PackageEntity(DiagramEntity):
     """A diagram object representing a package."""
 
+    default_shape = "package"
+
 
 class ClassEntity(DiagramEntity):
     """A diagram object representing a class."""
 
-    def __init__(self, title, node):
+    default_shape = "class"
+
+    def __init__(self, title: str, node: ClassDef) -> None:
         super().__init__(title=title, node=node)
-        self.attrs = None
-        self.methods = None
+        self.attrs: list[str] = []
+        self.methods: list[FunctionDef] = []
 
 
 class ClassDiagram(Figure, FilterMixIn):
     """Main class diagram handling."""
 
     TYPE = "class"
 
-    def __init__(self, title, mode):
+    def __init__(self, title: str, mode: str) -> None:
         FilterMixIn.__init__(self, mode)
         Figure.__init__(self)
         self.title = title
-        self.objects = []
-        self.relationships = {}
-        self._nodes = {}
-        self.depends = []
+        self.objects: list[Any] = []
```

Comment:
Not putting `Any` here would make `mypy` complain in `writer.py`. This can be resolved after the refactoring is done.

Reply:
Let's put a `TODO` here so we don't forgot about it.

```suggestion
        # TODO: Specify 'Any' after refactor of `DiagramEntity`
        self.objects: list[Any] = []
```

Reply:
Right, the `fixme` check was disabled for our codebase some days ago, right? So we can now use it without an `pylint disabe` now. 👍

Reply:
Yes!